src/core/Helios.DedicatedThreadPool/Helios.Concurrency.DedicatedThreadPool.cs

```diff
@@ -194,20 +194,54 @@ private void RequestWorker()
         }
     }
 
+
+
     /// <summary>
     /// An instanced, dedicated thread pool.
     /// </summary>
     internal class DedicatedThreadPool : IDisposable
     {
+        internal class DedicatedThreadPoolSupervisor
+        {
+            internal DedicatedThreadPoolSupervisor(DedicatedThreadPool pool)
+            {
+                //TODO: when should timer be killed?
+                var timer = new Timer(_ =>
+                {
+                    foreach (var worker in pool.Workers)
+                    {
+                        var w = worker;
+                        if (Interlocked.Exchange(ref w.Status, 0) == 0)
+                        {
+                            //this requests a new new worker and calls ForceTermination on the old worker
+                            //Potential problem here: if the thread is not dead for real, we might abort real work.. there is no way to tell the difference between 
+                            //deadlocked or just very long running tasks
+                            var newWorker = pool.RequestThread(w);
+                            continue;
+                        }
+
+                        //schedule heartbeat action to worker
+                        worker.AddWork(() =>
+                        {
+                            Interlocked.Increment(ref w.Status);
+                        });
+
+                    }
+                    //TODO: make the poll time out configurable
+                }, null, TimeSpan.FromSeconds(5), TimeSpan.FromSeconds(5));
+            }
+        }
        
         public DedicatedThreadPool(DedicatedThreadPoolSettings settings)
         {
             Settings = settings;
-            Workers = Enumerable.Repeat(new WorkerQueue(), settings.NumThreads).ToArray();
```

Comment:
This code will repeat the _same_ worker queue NumThread times.
The intention is to create a new worker queue for each repetition. this have been fixed